src/htm/engine/YAMLUtils.cpp

```diff
@@ -67,7 +67,13 @@ static void _toScalar(const YAML::Node &node, std::shared_ptr<Scalar> &s) {
     s->value.real64 = node.as<Real64>();
     break;
   case NTA_BasicType_Bool:
-    s->value.boolean = node.as<bool>();
+    try {
+      s->value.boolean = node.as<bool>(); // looking for 'true' or 'false'
+    } catch (...) {
+      int x = node.as<int>(); // looking for '0' or non-zero
+      s->value.boolean = (x == 0) ? false : true;
```

Comment:
> when does this happen?    (when is a boolean value parsed as "0" or "1" rather than "true" or "false")

I ran into this when trying to create the spec from the default values extracted from the encoder.  The macro is extracting the default value as binary value. The spec wants the default value as a JSON string.  The macro uses to_string() which returns "0" rather than "false".